jetty-servlets/src/main/java/org/eclipse/jetty/servlets/DoSFilter.java

```diff
@@ -1369,10 +1406,45 @@ public void onTimeout(AsyncEvent event) throws IOException
         }
     }
 
-    private String createRemotePortId(ServletRequest request)
+    public enum Action
     {
-        String addr = request.getRemoteAddr();
-        int port = request.getRemotePort();
-        return addr + ":" + port;
+        NO_ACTION,
+        REJECT,
+        DELAY,
+        THROTTLE;
```

Comment:
I still think we should add ABORT.... maybe even a DELAY_THEN_ABORT action.
```suggestion
        NO_ACTION,
        ABORT,
        REJECT,
        DELAY,
        THROTTLE;
```